### PR TITLE
feat: add telemetry documentation

### PR DESCRIPTION
**Pull request by SgtPooki**
fixes https://github.com/ipfs-shipyard/ignite-metrics/issues/29
fixes https://github.com/ipfs/ipfs-gui/issues/125

note that i’ve made some changes to the groupings of metric features:

removed groups: “tracking”, “marketing”
moved items: “events” is now in “minimal” metrics
new groups: “UX” (scrolls, clicks, forms), “Feedback” (star-rating, feedback), “location” (location)
removed items: “attribution”, “users”

cc @BigLep @lidel @tinytb @juliaxbow

Callouts for this PR: This PR is intended to close out https://github.com/ipfs-shipyard/ignite-metrics/issues/29 & https://github.com/ipfs/ipfs-gui/issues/125

Please bring up any concerns you have that I missed

### CODE REVIEW

File: src/MetricsProvider.ts
Comment (djmcquillan) [inline src/MetricsProvider.ts:29]: nit: do we need locationFeatures if we do not plan to collect this data? unless I am misinterpreting the code it might confuse users if it is here and we are not collecting any info.

File: docs/telemetry/CollectedData.template.md
Comment (SgtPooki) [reply docs/telemetry/CollectedData.template.md:21]: image

File: docs/telemetry/CollectedData.template.md
Comment (SgtPooki) [reply docs/telemetry/CollectedData.template.md:11]: image

I want to keep the exact ordering as the other section. How do you feel about `What metrics data we do NOT collect` ?

File: docs/telemetry/CollectedData.template.md
Comment (SgtPooki) [inline docs/telemetry/CollectedData.template.md:36]: This will need to point to `main/` prior to merging

File: docs/telemetry/CollectedData.template.md
Comment (SgtPooki) [inline docs/telemetry/CollectedData.template.md:3]: update URL to use `main` branch prior to merging

File: docs/telemetry/CollectionPolicy.md
Comment (SgtPooki) [reply docs/telemetry/CollectionPolicy.md:114]: I would rather be more explicit. we can mention PII data if we want.. what about `Are you trying to collect user-identifiable, or other user data at all? (i.e. PII data)`

File: docs/telemetry/FAQs.md
Comment (SgtPooki) [reply docs/telemetry/FAQs.md:3]: Do people frequently ask us why _they_ are collecting metrics? ➿ 😵 😛 

I'm fine with changing this, but also wondering... are there any guides on what point of view FAQs should be written in? I've never checked..

**Comment by BigLep** [comment]
I got to talk with @SgtPooki and @juliaxbow on 2023-01-06 and relay my mental model of:
1. reducing complexity for implementation/testing/maintenance
2. generally maximizing the amount of cases where we're collecting data (so we can be knowledgable about usage) while also not obsessing over it
3. be able to look at user in the face and feel good about the decisions that are being made

More of my specific comments were captured in that verbal conversation and in https://www.notion.so/pl-strflt/Telemetry-b005d4f217f44db3986902c67d922cf4

I expect to be fully tied up the week of 2023-01-09 and I don't want this work to block on me.  I believe I got my perspective across verbally and at this point the only "sign off" I want to make sure we have is from @lidel .  (I'd encourage a sync chat the week of 2023-01-09 so can close out on decisions quickly.).   I( didn't read this PR - only https://www.notion.so/pl-strflt/Telemetry-b005d4f217f44db3986902c67d922cf4 )

Summary: I'm not a blocking reviewer, but I would like @lidel to be.

Thanks all for navigating here.

File: docs/telemetry/PrivacyPolicy.md
Comment (lidel) [inline docs/telemetry/PrivacyPolicy.md:5]: lol, https://protocol.ai/legal/#privacy-policy has no content – it is a copy-paste of terms of service above about CC license, not a single word about privacy :upside_down_face: 


ps. for the future, no reason to use native URI like this. Many people won't have `ipns://` handler and won't be able to read this.
Just use normal URL/gateway one, and users with Brave/Copmpanion will be able to load it over IPFS anyway.

```suggestion
Until then, for all project's other than ipfs-companion (which is the only project we own with a current [privacy policy](https://github.com/ipfs/ipfs-companion/blob/main/PRIVACY-POLICY.md)), you can reference [protocol.ai/legal/#privacy-policy](https://protocol.ai/legal/#privacy-policy)
```

File: docs/telemetry/CollectedData.template.md
Comment (lidel) [inline docs/telemetry/CollectedData.template.md:19]:
```suggestion
Telemetry is sent to Countly instance at `countly.ipfs.tech`. You can read how they protect your data at [Security, Privacy, and Access FAQ](https://support.count.ly/hc/en-us/articles/360037501372-Security-Privacy-and-Access-FAQ)
```

File: docs/telemetry/CollectedData.template.md
Comment (lidel) [reply docs/telemetry/CollectedData.template.md]: (just an idea for the future)
We could replace these with aliases at `ipfs.fyi` (https://github.com/ipfs/ipfs.fyi):

```
/telemetry-collection-policy https://github.com...CollectionPolicy.md
/telemetry-privacy-policy    https://github.com...PrivacyPolicy.md
/telemetry-faq               https://github.com...FAQs.md
```

File: docs/telemetry/CollectionPolicy.md
Comment (lidel) [inline docs/telemetry/CollectionPolicy.md:3]: I am afraid user will have no idea what Ignite is.

```suggestion
This document is the source of truth for our telemetry policies and practices across all Ignite team (IPFS GUI) projects. The discussion held at https://github.com/ipfs/ipfs-gui/issues/125 is the origin and inspiration for this library and all documents under `./docs/telemetry`.
```

**Review by lidel — APPROVED** [review]
We've discussed this plan during a call today, and the proposed telemetry gathering focused on feature usage metrics is sensible. As [noted before](https://github.com/ipfs/ipfs-gui/issues/125#issuecomment-1352295236), I am okay with opt-out telemetry (like Firefox) as long we have a Privacy Policy and :+1:  from Legal.

Fine to merge this PR as documentation to unblock things like https://github.com/ipfs/ipfs-companion/pull/1117

Noting the small caveat I mentioned during the call: 
- We can't ship telemetry enabled by default in webui/desktop without having a privacy policy document first. 
  - :point_right:  Right now `docs/telemetry/PrivacyPolicy.md` is missing, and you refer to https://protocol.ai/legal/#privacy-policy which contains no privacy policy :upside_down_face:  :shrug:  (not blocking this PR, just flagging)

**Comment by SgtPooki** [comment]
@lidel 

> We've discussed this plan during a call today, and the proposed telemetry gathering focused on feature usage metrics is sensible. As [noted before](https://github.com/ipfs/ipfs-gui/issues/125#issuecomment-1352295236), I am okay with opt-out telemetry (like Firefox) as long we have a Privacy Policy and 👍 from Legal.

We got the :+1: from legal to proceed with opt-out metrics for all projects. We _will_ have a privacy policy from legal but do not have one yet (ETA was 1-2 weeks last week)

> Fine to merge this PR as documentation to unblock things like [ipfs/ipfs-companion#1117](https://github.com/ipfs/ipfs-companion/pull/1117)
> 
> Noting the small caveat I mentioned during the call:
> 
> * We can't ship telemetry enabled by default in webui/desktop without having a privacy policy document first.

I must have misheard during the call. Where is this constraint coming from? Legal explicitly told us gathering app metrics is not blocked by having a privacy policy.

>   * 👉  Right now `docs/telemetry/PrivacyPolicy.md` is missing, and you refer to https://protocol.ai/legal/#privacy-policy which contains no privacy policy 🙃  🤷  (not blocking this PR, just flagging)

I want to unblock the work required for webui & desktop. How would you recommend I update the content so that the following constraints are met:

1. Telemetry changes intended for webui & desktop are unblocked
2. PrivacyPolicy is an obvious placeholder until we get one from Legal

Two potential options:

1. Referencing the PrivacyPolicy in ipfs-companion (since Legal gave the go-ahead on changes to ipfs-companion without an update to its privacy policy)
2. Copying the PrivacyPolicy from ipfs-companion, with a note at the top mentioning that it was copied and will be updated soon.

**Comment by lidel** [comment]
>> We can't ship telemetry enabled by default in webui/desktop without having a privacy policy document first.
> Where is this constraint coming from? Legal explicitly told us gathering app metrics is not blocked by having a privacy policy.

I was under the impression that we will link to Privacy Policy and Telemetry Collection Policy documents on  the Settings screen near the opt-out button, so the user can read them before deciding to opt-out. 

If Legal said shipping without Privacy Policy is fine, and proper Privacy Policy document will land eventually, then a placeholder  referencing companion policy for now should be good enough :+1:

File: docs/telemetry/CollectionPolicy.md
Comment (SgtPooki) [reply docs/telemetry/CollectionPolicy.md:83]: The screenshots are slightly out of date and the wording in the documents in this PR are what we'll be using.

File: docs/telemetry/PrivacyPolicy.md
Comment (SgtPooki) [reply docs/telemetry/PrivacyPolicy.md:5]: resolved locally with update to ipfs-companion policy pushing up shortly.

File: docs/telemetry/FAQs.md
Comment (SgtPooki) [reply docs/telemetry/FAQs.md:34]: they won't be next to those other files because they'll be in `./docs/telemetry`.  Also, patterns of bad practice are a reason to stray, not stay, the course.

Note that ipfs-companion also breaks the convention for special files because it is UPPER-KEBAB-CASE, not UPPER_SNAKE_CASE. Also, PRIVACY_POLICY.md does not seem to be a recognized special file. See https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/about-community-profiles-for-public-repositories. There is an argument to be made for specifically not following the convention in this case so we do not conflate implications.

Special files are UPPER_SNAKE_CASE, but because these files are specific to the patterns outlined in this PR, I thought camel case was ideal. I'm fine to switch these to follow the standard for github special files if desired.

Comment (SgtPooki) [reply docs/telemetry/FAQs.md]: updated the docs to use UPPER_SNAKE_CASE

File: docs/telemetry/CollectedData.template.md
Comment (SgtPooki) [reply docs/telemetry/CollectedData.template.md]: > (just an idea for the future) We could replace these with aliases at `ipfs.fyi` (https://github.com/ipfs/ipfs.fyi):
> 
> ```
> /telemetry-collection-policy https://github.com...CollectionPolicy.md
> /telemetry-privacy-policy    https://github.com...PrivacyPolicy.md
> /telemetry-faq               https://github.com...FAQs.md
> ```

I love the idea but I'm not sure exactly how this works and there doesn't seem to be enough info in the readme. Are we supposed to create a _redirects file in each repo, or just in this repo?

File: docs/telemetry/CollectionPolicy.md
Comment (SgtPooki) [reply docs/telemetry/CollectionPolicy.md]: That's one good reason. But generally, it was marked as opt-in because I'm trying to keep the amount of opt-out metric groupings small to reduce risk for ourselves and our users. 

I've since switched all the groupings to "opt-out" or "N/A" with a note:

> **Note:** The above groupings are our global settings. You will want to check the project specific `./docs/telemetry/COLLECTED_DATA.md` for variations.

File: docs/telemetry/CollectionPolicy.md
Comment (SgtPooki) [reply docs/telemetry/CollectionPolicy.md]: i've switched it to opt out (see other comment) and will have this enabled by default in desktop & webui but not in other projects.

**Comment by SgtPooki** [comment]
> I was under the impression that we will link to Privacy Policy and Telemetry Collection Policy documents on the Settings screen near the opt-out button, so the user can read them before deciding to opt-out.

@lidel cc @juliaxbow @whizzzkid 

We can add links to both of those if we feel it's necessary, but I would prefer a single location to direct users to, such as the COLLECTION_POLICY in this repo, which links to the PRIVACY_POLICY. Does it make sense to have a separate landing page for users that's more concise? i.e. `./docs/telemetry/README.md` ?

File: docs/telemetry/CollectionPolicy.md
Comment (SgtPooki) [reply docs/telemetry/CollectionPolicy.md]: https://github.com/ipfs-shipyard/ipfs-css/issues/67

**Pull request merged**
